server/services/user.js

```diff
@@ -29,7 +30,7 @@ function updateUser(userId, userObject) {
     email: Joi.string().email(),
     auth: Joi.string(),
     groups: Joi.array().items(Joi.string().hex().length(24)),
-    devices: Joi.array(), // TODO this needs to be updated once the device model is done
+    devices: Joi.array().items(DEVICE_SCHEMA).options({ allowUnknown: true }),
```

Comment:
Would you be satisfied with `stripUnknown`? We're passing in a mongo object here, which has a bunch of keys under the hood.